src/dmd/mtype.d

```diff
@@ -4656,6 +4656,15 @@ extern (C++) final class TypeFunction : TypeNext
         return buf.extractString();
     }
 
+    private const(char)* getMatchError(A...)(const(char)* format, A args)
```

Comment:
The frontend doesn't support mangling that sort of template in C++. As it's private anyway, you should explicitly mark it as extern(D)